Add labels and annotations flags to config options in backend reference (#2384)
sensu/sensu-docs: content/sensu-go/5.20/reference/backend.md

### General configuration flags

| annotations|      |
-------------|------

> description  | Non-identifying metadata to include with event data for backend assets (e.g. handler and mutator assets).

Reviewer: I'm wondering if entity would make more sense here, because the event does not get altered with backend entity data. WDYT?
```suggestion
description  | Non-identifying metadata to include with entity data for backend assets (e.g. handler and mutator assets).
```

Author: Yes, I agree -- according to the notes in https://github.com/sensu/sensu-go/issues/3698 it sounds like entity is correct and not event. Thank you.

> description  | Non-identifying metadata to include with event data that you can access with [event filters][9] and [tokens][27]. You can use annotations to add data that is meaningful to people or external tools that interact with Sensu.<br><br>In contrast to labels, you cannot use annotations in [API response filtering][25], [sensuctl response filtering][26], or [dashboard view filtering][54].

Reviewer: Reading this information back I'm realizing that most of this is only applicable to the agent entity doc. Backend entity labels and annotations will only be used in filters and token substitution (not yet implemented) for backend assets (ex. handler assets and mutator assets). API response filtering, sensuctl response filtering, and dashboard view filtering will not be possible for either field because we do not currently store the backend as an entity.

> description  | Non-identifying metadata to include with event data that you can access with [event filters][9] and [tokens][27]. You can use annotations to add data that is meaningful to people or external tools that interact with Sensu.<br><br>In contrast to labels, you cannot use annotations in [API response filtering][25], [sensuctl response filtering][26], or [dashboard view filtering][54].

Author: ```suggestion
description  | Non-identifying metadata to include with event data for backend assets (e.g. handler and mutator assets).
```

required     | false
type         | Map of key-value pairs. Keys and values can be any valid UTF-8 string.
default      | `null`
environment variable | `SENSU_ANNOTATIONS`
example      | {{< highlight shell >}}# Command line examples

> sensu-backend start --annotations sensu.io/plugins/slack/config/webhook-url=https://hooks.slack.com/services/T00000000/B00000000/XXXXXXXXXXXXXXXXXXXXXXXX

Reviewer: This example is fine as is, since a slack webhook in a handler asset is a valid use case.

deregistration-handler: "/path/to/handler.sh"{{< /highlight >}}

| labels     |      |
-------------|------

> description  | Custom attributes to include with event data for backend assets (e.g. handler and mutator assets).

Reviewer: I'm wondering if entity would make more sense here, because the event does not get altered with backend entity data. WDYT?
```suggestion
description  | Custom attributes to include with entity data for backend assets (e.g. handler and mutator assets).
```

> description  | Custom attributes to include with event data that you can use for response and dashboard view filtering.<br><br>If you include labels in your event data, you can filter [API responses][25], [sensuctl responses][26], and [dashboard views][54] based on them. In other words, labels allow you to create meaningful groupings for your data.<br><br>Limit labels to metadata you need to use for response filtering. For complex, non-identifying metadata that you will *not* need to use in response filtering, use annotations rather than labels.

Reviewer: Response filtering not possible, this should all probably be removed.

> description  | Custom attributes to include with event data that you can use for response and dashboard view filtering.<br><br>If you include labels in your event data, you can filter [API responses][25], [sensuctl responses][26], and [dashboard views][54] based on them. In other words, labels allow you to create meaningful groupings for your data.<br><br>Limit labels to metadata you need to use for response filtering. For complex, non-identifying metadata that you will *not* need to use in response filtering, use annotations rather than labels.

Author: ```suggestion
description  | Custom attributes to include with event data for backend assets (e.g. handler and mutator assets).
```

required     | false
type         | Map of key-value pairs. Keys can contain only letters, numbers, and underscores and must start with a letter. Values can be any valid UTF-8 string.
default      | `null`
environment variable | `SENSU_LABELS`
example               | {{< highlight shell >}}# Command line examples

> sensu-backend start --labels proxy_type=website

Reviewer: This `proxy_type` example makes sense as an agent label, but not really as a backend label. I would suggest a different example such as `security_zone=us-west-2a` or something more applicable to a real-world use case of a backend entity label.

> sensu-backend start --labels proxy_type=website

Author: ```suggestion
sensu-backend start --labels security_zone=us-west-2a
```

sensu-backend start --labels example_key1="example value" example_key2="example value"

# /etc/sensu/backend.yml example
labels:

>   proxy_type: "website"

Author: ```suggestion
  security_zone: "us-west-2a"
```
